easybuild/tools/options.py

```diff
@@ -504,36 +511,126 @@ def avail_cfgfile_constants(self):
+            table_line = line_tmpl.format('', '', '', c='=', nw=nw, dw=dw, dfw=dfw)
+
+            # table header
+            lines.append(table_line)
+            lines.append(line_tmpl.format(name_title, descr_title, dflt_title, c=' '))
+            lines.append(line_tmpl.format('', '', '', c='-'))
+
+            # table rows by parameter
+            for name, (descr, dflt) in sorted(grouped_params[grpname].items()):
+                rawname = '``%s``' % name
+                lines.append(line_tmpl.format(rawname, descr, str(quote_str(dflt)), c=' '))
+            lines.append(table_line)
+            lines.append('')
+
+        return '\n'.join(lines)
+
     def avail_easyconfig_params(self):
         """
-        Print the available easyconfig parameters, for the given easyblock.
+        Compose overview of available easyconfig parameters, in specified format.
         """
-        extra = []
+        params = copy.deepcopy(DEFAULT_CONFIG)
+
+        # include list of extra parameters (if any)
+        extra_params = {}
         app = get_easyblock_class(self.options.easyblock, default_fallback=False)
         if app is not None:
-            extra = app.extra_options()
-        mapping = convert_to_help(extra, has_default=False)
-        if extra:
-            ebb_msg = " (* indicates specific for the %s EasyBlock)" % app.__name__
-            extra_names = [x[0] for x in extra]
-        else:
-            ebb_msg = ''
-            extra_names = []
-        txt = ["Available easyconfig parameters%s" % ebb_msg]
-        params = [(k, v) for (k, v) in mapping.items() if k.upper() not in ['HIDDEN']]
-        for key, values in params:
-            txt.append("%s" % key.upper())
-            txt.append('-' * len(key))
-            for name, value in values:
-                tabs = "\t" * (3 - (len(name) + 1) / 8)
-                if name in extra_names:
-                    starred = '(*)'
-                else:
-                    starred = ''
-                txt.append("%s%s:%s%s" % (name, starred, tabs, value))
-            txt.append('')
-
-        return "\n".join(txt)
+            extra_params = app.extra_options()
+        if isinstance(extra_params, ExtraOptionsDeprecatedReturnValue):
+            extra_params = dict(extra_params)
+        params.update(extra_params)
+
+        # compose title
+        title = "Available easyconfig parameters"
+        if extra_params:
+            title += " (* indicates specific to the %s easyblock)" % app.__name__
+
+        # group parameters by category
+        grouped_params = OrderedDict()
+        for category in sorted_categories():
+            # exclude hidden parameters
+            if category[1].upper() in [HIDDEN]:
+                continue
+
+            grpname = category[1]
+            grouped_params[grpname] = {}
+            for name, (dflt, descr, cat) in params.items():
+                # FIXME bug in default.py?
```

Comment:
yeah, I should, I included this as a sort of placeholder...

The problem is that, for example, the constant `CUSTOM` as it is used in `framework/easyconfig/default.py`  corresponds to the string `"CUSTOM"`. 
However, the constant `CUSTOM` that is imported from `framework.easyconfig`, as is done in easyblocks when defining custom parameters via `extra_options`, corresponds to `(1, 'easyblock-specific')`, since it's obtained from `ALL_CATEGORIES` in framework/easyconfig/__init__.py`.

That causes problems when comparing categories, as is done here, since then all categories should be of the same kind, whether they are obtained for default or extra parameters.

I can probably fix it without breaking easyblocks, but I need to think about it. :-)